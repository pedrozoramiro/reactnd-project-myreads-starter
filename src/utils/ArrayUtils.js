export function mergeByProperty(target, source,property) {
  for (let item of source) {
    let existingItem = target.find(element => element[property] === item[property]);
    if (existingItem) {
      Object.assign(existingItem, item);
      continue;
    }
    target.push(item);
  }
  return target;
}