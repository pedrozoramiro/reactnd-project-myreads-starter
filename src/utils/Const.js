export const SHELVES= [
    { name: "currentlyReading", title: "Currently Reading" },
    { name: "wantToRead", title: "Want to Read" },
    { name: "read", title: "Read" }
  ];