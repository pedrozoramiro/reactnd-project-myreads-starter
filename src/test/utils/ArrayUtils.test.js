
import * as ArrayUtils from "../../utils/ArrayUtils";

it("Should merge property", () => {
    const targetArray = [{id:'1',name:'name1'},{id:'2',name:'name2'}] 
    const mergeArray = [{id:'1',name:'merged1'}] 
    const  merged = ArrayUtils.mergeByProperty(targetArray,mergeArray,'id');
    expect(merged).toMatchSnapshot();
  });

  it("Should add new item ", () => {
    const targetArray = [{id:'1',name:'name1'},{id:'2',name:'name2'}] 
    const mergeArray = [{id:'3',name:'name3'}] 
    const  merged = ArrayUtils.mergeByProperty(targetArray,mergeArray,'id');
    expect(merged).toMatchSnapshot();
  });