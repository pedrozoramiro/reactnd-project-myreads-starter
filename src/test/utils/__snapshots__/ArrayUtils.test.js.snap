// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should add new item  1`] = `
Array [
  Object {
    "id": "1",
    "name": "name1",
  },
  Object {
    "id": "2",
    "name": "name2",
  },
  Object {
    "id": "3",
    "name": "name3",
  },
]
`;

exports[`Should merge property 1`] = `
Array [
  Object {
    "id": "1",
    "name": "merged1",
  },
  Object {
    "id": "2",
    "name": "name2",
  },
]
`;
